staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt.go

```diff
@@ -164,33 +177,23 @@ type requestManagement struct {
 	requestWaitLimit time.Duration
 
 	// curState holds a pointer to the current RMState.  That is,
-	// `Load()` produces a `*RMState`.  When a config work queue worker
-	// processes a configuration change, it stores a new pointer here ---
-	// it does NOT side-effect the old `RMState` value.  The new `RMState`
-	// has a freshly constructed slice of FlowSchema pointers and a
-	// freshly constructed map of priority level states.  But the new
-	// `RMState.priorityLevelStates` includes in its range at least all the
-	// `*PriorityLevelState` values of the old `RMState.priorityLevelStates`.
-	// Consequently the filter can load a `*RMState` and work with it
-	// without concern for concurrent updates.  When a priority level is
-	// finally deleted, this will also involve storing a new `*RMState`
-	// pointer here, but in this case the range of the
-	// `RMState.priorityLevels` will be reduced --- by removal of the
-	// priority level that is no longer in use.
+	// `Load()` produces a `*RMState`.  When a config work queue
+	// worker processes a configuration change, it stores a new
+	// pointer here --- it does NOT side-effect the old `RMState`
+	// value.  The new `RMState` has a freshly constructed slice of
+	// FlowSchema pointers and a freshly constructed map of pointers
+	// to immutable PriorityLevelState values.  The new
+	// `RMState.priorityLevelStates` includes in its domain the names
+	// of all the current and lingering priority levels.  Consequently
+	// the filter can load a `*RMState` and work with it without
+	// concern for concurrent updates.  When a priority level is
+	// finally removed from this server, this involves storing a new
+	// `*RMState` pointer here and in this case the domain of the
+	// `RMState.priorityLevels` omits the name of the priority level
+	// being removed.
 	curState atomic.Value
 }
 
-// TypedConfigObjectReference is a reference to a relevant config API object.
-// No namespace is needed because none of these objects is namespaced.
-type TypedConfigObjectReference struct {
-	Kind string
-	Name string
-}
-
-func (tr *TypedConfigObjectReference) String() string {
-	return tr.Kind + "/" + tr.Name
-}
-
 // rmSetup is invoked at startup to create the infrastructure of this filter
 func rmSetup(kubeClient kubernetes.Interface, serverConcurrencyLimit int, requestWaitLimit time.Duration, clk clock.Clock) *requestManagement {
 	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, 0)
```

Comment:
should not instanciate a new informer factory here, there should be only one factory working in one apiserver.

Reply:
I think that's probably true in a main apiserver; is it also true in an extension server?

How would this code get passed that factory?